foundationdb/datadog_checks/foundationdb/check.py

```diff
@@ -0,0 +1,219 @@
+from datadog_checks.base import AgentCheck
+import json
+from typing import Any
+import fdb
+
+fdb.api_version(600)
+
+class FoundationdbCheck(AgentCheck):
+    def __init__(self, name, init_config, instances):
+        super(FoundationdbCheck, self).__init__(name, init_config, instances)
+        self.hasDB = False
```

Comment:
WDYTA caching? https://github.com/DataDog/integrations-core/pull/8083

e.g.

```python

    def __init__(self, name, init_config, instances):
        super(FoundationdbCheck, self).__init__(name, init_config, instances)
        ...
        self.db = None

    @property
    def db(self):
        if self.db is None:
            self.db = ...

        return self.db
```

Reply:
Hm, we did something like this but now I'm reviewing it and I'm confused how the Oracle one you linked is correct. It's constructed like this:

```
    def __init__(self, name, init_config, instances):
        super(Oracle, self).__init__(name, init_config, instances)
```

That is, with many instances, but then goes on to do:

```
        self._server = self.instance.get('server')
        self._user = self.instance.get('user')
```

But...which instance? If we were just constructed with all of them, how can we assume there's one of them?

Reply:
Ah, OK, good. Then I'm comfortable what we're doing (caching the connection on the instance) will work.